timelock-impl/src/test/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResourceTest.java

```diff
@@ -91,6 +105,41 @@ public void requestThrowsIfAnyQueryFails() {
                 .isInstanceOf(BlockingTimeoutException.class);
     }
 
+    @Test
+    public void canStartTransactionsForMultipleClients() {
```

Comment:
Nice!

However, it's probably also worth having a test where the requests and responses are different for each namespace (even if as simple as different number of requests). Then you can verify that different requests and responses are actually being sent / received.